huaweicloud/services/er/data_source_huaweicloud_er_availability_zones.go

```diff
@@ -0,0 +1,130 @@
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func DataSourceAvailabilityZones() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceAvailabilityZonesRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			// Attributes
+			"availability_zones": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"code": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"state": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAvailabilityZonesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		listAvailabilityZoneHttpUrl = "v3/{project_id}/enterprise-router/availability-zones"
+		listAvailabilityZoneProduct = "er"
+	)
+	listAvailabilityZoneClient, err := cfg.NewServiceClient(listAvailabilityZoneProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating ER client: %s", err)
+	}
+
+	listAvailabilityZonePath := listAvailabilityZoneClient.Endpoint + listAvailabilityZoneHttpUrl
+	listAvailabilityZonePath = strings.ReplaceAll(listAvailabilityZonePath, "{project_id}",
+		listAvailabilityZoneClient.ProjectID)
+	listAvailabilityZonequeryParams := buildListAvailabilityZonesQueryParams(d)
+	listAvailabilityZonePath += listAvailabilityZonequeryParams
+
+	listAvailabilityZoneOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	listAvailabilityZoneResp, err := listAvailabilityZoneClient.Request("GET", listAvailabilityZonePath,
+		&listAvailabilityZoneOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	listAvailabilityZoneRespBody, err := utils.FlattenResponse(listAvailabilityZoneResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+
+	var mErr *multierror.Error
+	mErr = multierror.Append(
+		mErr,
+		d.Set("availability_zones", flattenListAvailabilityZone(listAvailabilityZoneRespBody)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenListAvailabilityZone(resp interface{}) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	curJson := utils.PathSearch("availability_zones", resp, make([]interface{}, 0))
```

Comment:
1. we should only return the AZs which state is **available**;
2. the schema can be defined as **names**, please refer to https://registry.terraform.io/providers/huaweicloud/huaweicloud/latest/docs/data-sources/availability_zones

```suggestion
	curJson := utils.PathSearch("availability_zones[?state == 'available'].code | sort(@)", resp, make([]string, 0))
```